core/subbin_sizes.cpp

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#include <algorithm>
+#include <numeric>
+
+#include "scipp/common/except.h"
+#include "scipp/core/element/arithmetic.h"
+#include "scipp/core/subbin_sizes.h"
+
+namespace scipp::core {
+
+SubbinSizes::SubbinSizes(const scipp::index offset, container_type &&sizes)
+    : m_offset(offset), m_sizes(std::move(sizes)) {}
+
+SubbinSizes::SubbinSizes(const scipp::index value)
+    : m_offset(0), m_sizes({value}) {}
+
```

Comment:
This is an internal type so, unlikely to get exposed to misuse. However, did you consider how this behaves with bad inputs negative offsets etc. I suspect checking would hit performance even if it was considered worthwhile.

Reply:
I think performance should not be affected much, I can add a check.